lib/task/import/importDeleteTask.class.php

```diff
@@ -0,0 +1,283 @@
+        ),
+      ]
+    );
+  }
+
+  /**
+   * @see sfTask
+   */
+  public function execute($arguments = array(), $options = array())
+  {
+    $count = 0;
+
+    parent::execute($arguments, $options);
+
+    if ($options['logfile'])
+    {
+      $this->registerFileLogger($options['logfile']);
+    }
+
+    if ($options['verbose'])
+    {
+      $this->verbose = true;
+    }
+
+    $this->getImportObjectIds($arguments['name']);
+
+    $this->confirmDeletion($options['force']);
+
+    $this->timer = new QubitTimer();
+
+    foreach ($this->objectIds as $id)
+    {
+      $this->deleteObject($id);
+
+      $this->deleteKeymapRow($arguments['name'], $id);
+
+      $count++;
+    }
+
+    $this->logMsg(
+      sprintf(
+        'Deleted %u objects from import "%s"', $count, $arguments['name']
+      )
+    );
+  }
+
+  public function logMsg($message, $priority = sfLogger::INFO, $timer = true)
+  {
+    if (sfLogger::DEBUG == $priority)
+    {
+      // If this is a debugging message, don't show it unless running in
+      // "verbose" mode
+      if (!$this->verbose)
+      {
+        return;
+      }
+    }
+
+    // No timing output
+    if (!$timer || null == $this->timer)
+    {
+      parent::log($message);
+
+      return;
+    }
+
+    // Include timing output
+    parent::log(
+      sprintf(
+        '[+%ss] %s ',
+        str_pad($this->timer->elapsed(), 7, '0', STR_PAD_LEFT),
+        $message
+      )
+    );
+  }
+
+  private function registerFileLogger($filename)
+  {
+    $flogger = new sfFileLogger($this->dispatcher, ['file' => $filename]);
+
+    $this->dispatcher->connect(
+      'command.log', array($flogger, 'listenToLogEvent')
+    );
+  }
+
+  private function getImportObjectIds($name)
+  {
+    // Get import rows from keymap table in reverse order, so objects imported
+    // last are deleted first (LIFO) to avoid parent_id constraint violations
+    $sql = <<<EOL
+SELECT target_id FROM keymap WHERE source_name=:name ORDER BY id DESC
+EOL;
+
+    $results = QubitPdo::fetchAll(
+      $sql, [':name' => $name], ['fetchMode' => PDO::FETCH_COLUMN]
+    );
+
+    if (count($results) < 1)
+    {
+      throw new sfException(
+        sprintf(
+          'No import data for import "%s" found in the keymap table', $name
+        )
+      );
+    }
+
+    $this->count = count($results);
+    $this->objectIds = $results;
+
+    $this->logMsg(
+      sprintf(
+        'Found %s resources created by import "%s"', $this->count, $name
+      )
+    );
+  }
+
+  /**
+   * Allow the user to bail out if they aren't sure they want to delete targeted
+   * descriptions.
+   *
+   * @param bool $isForced true to force delete without confirmation
+   *
+   * @return bool true to abort script, false to continue
+   */
+  private function confirmDeletion($isForced)
+  {
+    if ($isForced)
+    {
+      return;
+    }
+
+    $confirmed = $this->askConfirmation(
+      [
+"Continuing will delete $this->count records and related data.",
```

Comment:
Nitpick: indenting (but maybe indenting would make lines too long in which case ignore this comment ;) ).